Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -93,33 +113,34 @@ private static LookupElement createLookupElement(Project project, CompletionSugg
         icon = applyOverlay(icon, element.isConst(), AllIcons.Nodes.FinalMark);
         lookup = lookup.withIcon(icon);
       }
-      // insert
-      // TODO(scheglov) insert () with arguments
-      //lookup = lookup.withInsertHandler(new InsertHandler<LookupElement>() {
-      //  @Override
-      //  public void handleInsert(InsertionContext context, LookupElement item) {
-      //    context.getDocument().insertString(context.getStartOffset(), "()");
-      //  }
-      //});
+      // insert handler
+      lookup = lookup.withInsertHandler(new InsertHandler<LookupElement>() {
+        @Override
+        public void handleInsert(InsertionContext context, LookupElement item) {
+          final TemplateManager templateManager = TemplateManager.getInstance(project);
+          final Template template = templateManager.createTemplate("", "");
+          // Fill the template with the parameter names.
+          final List<String> parameterNames = suggestion.getParameterNames();
+          if (parameterNames != null) {
```

Comment:
it seems to be more accurate to check `if (parameterNames != null)` earlier and simply not to add insertHandler if not needed